terms/confusionmatrix_df.tex

```diff
@@ -0,0 +1,199 @@
+  \label{tab:cm_sweets}
+\end{table}
+
+The CM of the model $h : X \mapsto C$ over the concept $c : X \mapsto C$ using dataset $S \subset X$ is formally defined \citep{cichosz2014data} as a matrix $\Xi$ such that $\Xi_{c,S}(h)[d_1,d_2] = |S_{h=d_1,c=d_2}|$. The CM is constructed by incrementing the element corresponding to the true class \textit{vis-a-vi} the predicted class for each object in the dataset (Algorithm \ref{alg:cm}).
+
+\begin{algorithm}
+  \begin{algorithmic}
+    \State $\Xi \gets 0$
+    \For{$x \in S$}
+      \State $d_1 \gets c(x)$
+      \State $d_2 \gets h(x)$
+      \State $\Xi_{d_1,d_2} \gets \Xi_{d_1,d_2} + 1$
+    \EndFor
+  \end{algorithmic}
+  \caption{The CM is initialised to the zero matrix, and populated by iterating over all the objects $x$ with corresponding true class $d_1$ and predicted class $d_2$ and incrementing the element $(d_1,d_2)$ by 1 for each matching outcome.}
+  \label{alg:cm}
+\end{algorithm}
+
+In binary classification, the CM consists of 2 specially designated classes called the \textit{positive} class and the \textit{negative} class \citep{saito2015precision}. As indicated in Table \ref{tab:cm_binary}, positive outcomes from the true class  which are classified correctly are called \textit{True Positives} (TP), whilst misclassifications are called \textit{False Negatives} (FN). On the other hand, negative true class outcomes which are classified correctly are called \textit{True Negatives} (TN), and misclassifications are called \textit{False Positives} (FP). In natural sciences, FP are called \textit{Type I Errors} and FN are known as \textit{Type II Errors} \citep{fielding1997review}.
+
+\begin{table}[ht]
+  \centering
+  \fontfamily{ppl}\selectfont
+  \begin{tabular}{lll}
+    \toprule
+                  & \textit{+ve} & \textit{-ve} \\
+    \midrule
+    \textit{+ve}  & TP           & FN \\
+    \textit{-ve}  & FP           & TN \\
+    \bottomrule
+  \end{tabular}
+  \caption{CMs of binary classifiers have positive (+ve) and negative (-ve) classes, and elements called \textit{True Positives} (TP), \textit{False Positives} (FP), \textit{True Negatives} (TN) and \textit{False Negatives} (FN).}
+  \label{tab:cm_binary}
+\end{table}
+
+The information presented in the CM can be used to evaluate the performance of different binary classifiers \citep{lu2004predicting}. A number of statistics (Eq. \ref{eq:cm_acc}-\ref{eq:cm_fscore}) derived from the CM have been proposed in the literature \citep{deng2016improved} to gain a better understanding of what are the strengths and weaknesses of different classifiers. Caution should be exercised when interpreting metrics \citep{jeni2013facing}, since the CM could be misleading if the data is imbalanced and an important subrange of the domain (e.g. a rare diseases) is underrepresented \citep{raeder2012learning}.
```

Comment:
use `e.g.\ ` so the space is not considered as one after the sentence.  also give an example of how imbalanced datasets can produce misleading accuracy